satpy/composites/__init__.py

```diff
@@ -402,7 +402,7 @@ def __call__(self, projectables, **info):
             self.coszen[key] = coszen
         elif coszen is None:
             # we were given the SZA, calculate the cos(SZA)
-            coszen = np.cos(np.deg2rad(projectables[1]))
+            coszen = np.cos(np.deg2rad(info['optional_datasets'][0]))
```

Comment:
Isn't it possible that `match_data_arrays` changed this optional dataset? In that case we need to use that modified version of it. Right?

Reply:
Good point, I hadn't realized `match_data_arrays` would change the data